ledger/acctupdates.go

```diff
@@ -1612,6 +1631,86 @@ func (au *accountUpdates) lookupWithRewards(rnd basics.Round, addr basics.Addres
 	}
 }
 
+// lookupFullWithRewards returns the full account data for a given address at a given round.
+// The rewards are added to the AccountData before returning. Note that the function doesn't update the account with the rewards,
+// even while it does return the AccoutData which represent the "rewarded" account data.
+func (au *accountUpdates) lookupFullWithRewards(rnd basics.Round, addr basics.Address) (pad ledgercore.PersistedAccountData, err error) {
```

Comment:
this would not work correctly - you need to make all the reads from the database within the same `RLock`. Taking the lock twice means that in-between there could be a database write.

Reply:
thanks, fill fix here and below.